### PR TITLE
WT-9475	Write tests for cursor next/prev implementation

### PR DESCRIPTION
**Pull request by clarissecheah**
- Test cursor bound API with history store scenarios, prepared transactions, and statistics checking.
- Added `test_cursor_bound08.py` to `test_cursor_bound10.py`.
- Added complex colgroups testing to some of the existing tests.
- Fixed bug entry counting inside cursor_bound_traversal

### CODE REVIEW

File: test/suite/test_cursor_bound02.py [hunk 1]
```diff
@@ -53,23 +53,31 @@ class test_cursor_bound02(bound_base):
         ('composite_complex', dict(key_format='iSru')),
     ]
 
+    value_formats = [
+        ('string', dict(value_format='S')),
+        ('complex-string', dict(value_format='SS')),
+    ]
+
```
Comment (luke-pearson) [inline test/suite/test_cursor_bound02.py:60]: Remove extra whitespace.

File: test/suite/test_cursor_bound05.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ class test_cursor_bound05(bound_base):
     file_name = 'test_cursor_bound05'
     key_format = 'S'
     start_key = 1000
-    end_key = 2000
+    end_key = 1999
```
Comment (luke-pearson) [inline test/suite/test_cursor_bound05.py:40]: This deserves a comment explaining why it is correct, I understand the issue but it does look odd at first glance.

File: test/suite/test_cursor_bound07.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
 class test_cursor_bound07(bound_base):
     file_name = 'test_cursor_bound07'
     start_key = 10
-    end_key = 100
+    end_key = 99
```
Comment (luke-pearson) [inline test/suite/test_cursor_bound07.py:38]: Same here.

File: test/suite/test_cursor_bound08.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+from wtbound import bound_base
+from wiredtiger import stat
+
+# test_cursor_bound08.py
+# Test that the statistics added for cursor bound API is appropiately incremented for all different cursor
```
Comment (luke-pearson) [inline test/suite/test_cursor_bound08.py:35]: Grammar: are appropriately incremented

File: test/suite/test_cursor_bound10.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    ]
+
+    direction = [
+        ('prev', dict(next=False)),
+        ('next', dict(next=True)),
+    ]
+
+    scenarios = make_scenarios(types, key_format_values, config, direction)
+
+    def create_session_and_cursor(self):
```
Comment (luke-pearson) [inline test/suite/test_cursor_bound10.py:65]: Can we move the create_session_and_cursor function to the wtbound class so we de-duplicate code a bit? I would also be interested in knowing if a SimpleDataSet can be used here but I doubt it handles complex key formats.

File: test/suite/test_cursor_bound03.py [hunk 6]
```diff
@@ -148,7 +128,7 @@ def test_bound_general_scenario(self):
         # Test bound api: Test upper bound clearing with only lower bounds.
         self.set_bounds(cursor, 50, "lower")
         cursor.bound("action=clear,bound=upper")
-        self.cursor_traversal_bound(cursor, None, None, self.direction, self.end_key - 50)
+        self.cursor_traversal_bound(cursor, None, None, self.direction, self.end_key - 50 + 1)
```
Comment (luke-pearson) [inline test/suite/test_cursor_bound03.py:131]: This is a bit odd and should have just be the actual value we subtract which is -49?

File: test/suite/test_cursor_bound09.py [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+        
+        try:
+            cursor2.next()
+        except WiredTigerError as e:
+            if wiredtiger_strerror(WT_PREPARE_CONFLICT) in str(e):
+                self.assertEqual(cursor2.get_key(),self.check_key(30))
+                pass
+            else:
+                raise e
+        session2.rollback_transaction()
```
Comment (luke-pearson) [inline test/suite/test_cursor_bound09.py:214]: File needs a newline at the end of it.

File: test/suite/test_cursor_bound08.py [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+    ]
+
+    evict = [
+        ('inclusive-evict', dict(evict=True)),
+        ('inclusive-no-evict', dict(evict=False)),
+    ]
+
+    scenarios = make_scenarios(types, key_format_values, evict)
+
+    def create_session_and_cursor_timestamp(self):
```
Comment (luke-pearson) [inline test/suite/test_cursor_bound08.py:67]: I take it we couldn't merge this with the wtbound class as it required timestamps?

File: test/suite/test_cursor_bound03.py (hunk 6 above)
Comment (jiechenbo) [reply test/suite/test_cursor_bound03.py:131]: Yeah thats fair, i guess I wanted to show that self.start_key is a key itself as well. But that is misleading here. I agree lets make it -49.

**Review by luke-pearson — APPROVED** [review]
LGTM, with a few remaining comments.

File: test/suite/test_cursor_bound08.py (hunk 8 above)
Comment (clarissecheah) [reply test/suite/test_cursor_bound08.py:67]: Yep, that's what happened.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**